pkg/controller/mongodb/mongodb_status_options.go

```diff
@@ -112,6 +115,10 @@ func (o *optionBuilder) withStatefulSetReplicas(members int) *optionBuilder {
 }
 
 func (o *optionBuilder) withMessage(severityLevel severity, msg string) *optionBuilder {
+	if apierrors.IsTransient(errors.New(msg)) {
```

Comment:
I don't know if I'm totally convinced of this flow:

`withMessage` receives a string, which then is used to create an `error`, from which then a `string` is extracted.

But I understand if this is intended in order to have `IsTransient` adaptable to other scenarios in which it might receive an actual `error`

Reply:
@bznein I agree it's not a very nice usage. We could do something like

`IsTransientError` and `IsTransientMessage` or similar.

What do you think?

Reply:
Yes I think it would be better :) Thanks!